src/rhos/plotutils.py

```diff
@@ -0,0 +1,325 @@
+        + r" $\;\;\; \sigma$"
+        + r" = {:2.2f} ".format(invLapW.par.sigma / invLapW.par.massNorm)
+        + r"$M_{\rm ref}$"
+    )
+    if invLapW.par.Na > 1:
+        _a0label = r"$\alpha = {:1.2f}$".format(invLapW.algorithmPar.alphaA)
+    elif invLapW.par.Na == 1:
+        _a0label = r'$\rho_\sigma$'
+
+    plotwErr(ax, invLapW.lambda_list[invLapW.espace_dictionary[estar]],
+             invLapW.rho_list[invLapW.espace_dictionary[estar]],
+             invLapW.errBoot_list[invLapW.espace_dictionary[estar]], markerId=0, colorID=0, label=_a0label)
+
+    if invLapW.par.Na > 1:
+        plotwErr(ax, invLapW.lambda_list[invLapW.espace_dictionary[estar]],
+                 invLapW.rho_list_alphaB[invLapW.espace_dictionary[estar]],
+                 invLapW.errBoot_list_alphaB[invLapW.espace_dictionary[estar]], markerId=1, colorID=1, label=r"$\alpha = {:1.2f}$".format(invLapW.algorithmPar.alphaB))
+        if invLapW.par.Na > 2:
+            plotwErr(ax, invLapW.lambda_list[invLapW.espace_dictionary[estar]],
+                     invLapW.rho_list_alphaC[invLapW.espace_dictionary[estar]],
+                     invLapW.errBoot_list_alphaC[invLapW.espace_dictionary[estar]], markerId=2, colorID=2, label=r"$\alpha = {:1.2f}$".format(invLapW.algorithmPar.alphaC))
+
+    ax.axhspan(
+        ymin=float(invLapW.rhoResultHLT[invLapW.espace_dictionary[estar]]
+                   - invLapW.drho_result[invLapW.espace_dictionary[estar]]),
+        ymax=float(invLapW.rhoResultHLT[invLapW.espace_dictionary[estar]]
+                   + invLapW.drho_result[invLapW.espace_dictionary[estar]]),
+        alpha=0.3,
+        color=CB_colors[4],
+    )
+    ax.set_xlabel(r"$\lambda$", fontsize=32)
+    ax.set_ylabel(r"$\rho_\sigma$", fontsize=32)
+    ax.legend(prop={"size": 26, "family": "Helvetica"}, frameon=False)
+    ax.set_xscale('log')
+    plt.tight_layout()
+    if savePlot == True:
+        plt.savefig(
+            os.path.join(
+                invLapW.par.plotpath,
+                "LambdaScanE{:2.2e}".format(invLapW.espace_dictionary[estar]) + ".png",
+            ),
+            dpi=420,
+        )
+    if plot_live == True:
+        plt.show()
+    plt.clf()
+    plt.close(fig)
+    return
+
+def sharedPlot_stabilityPlusLikelihood(invLapW, estar, savePlot=True, plot_live=False):
+
+    setPlotOpt(plt)
+    fig, (ax, ax2) = plt.subplots(nrows=2, sharex=True, figsize=(8, 10), gridspec_kw={'height_ratios': [3, 2]})
+
+    plotwErr(ax, invLapW.lambda_list[invLapW.espace_dictionary[estar]],
+            invLapW.rho_list[invLapW.espace_dictionary[estar]],
+            invLapW.errBayes_list[invLapW.espace_dictionary[estar]], label='Bayesian Error', markerId=0, colorID=0)
+
+    plotwErr(ax, invLapW.lambda_list[invLapW.espace_dictionary[estar]],
+            invLapW.rho_list[invLapW.espace_dictionary[estar]],
+            invLapW.errBoot_list[invLapW.espace_dictionary[estar]], label='Bootstrap Error', markerId=1, colorID=1)
+
+    if(0):
```

Comment:
Not sure if this has been moved in from other code, but commenting in case I don't see it again and forget about it:

 - In the worst case, `if(0):` should be `if False:`
 - But really, there shouldn't be an unreachable block.
 - The contents of this block is a string, so even without the `if(0)` it will still do nothing
 - But in general multiline strings shouldn't be used for block comments (your editor should have a "comment region" function)
 - Large blocks of commented code should be avoided in general—we use version control to be able to retrieve previous versions of code.

Reply:
I have removed that one, as it seemed to me to be not quite relevant. If @LupoA  disagrees, I can restore it and have a conversation about this one.

Reply:
sweet!